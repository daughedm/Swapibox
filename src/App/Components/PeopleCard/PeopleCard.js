import React from 'react';
import '../PeopleCard/PeopleCard.css';

const PeopleCard = ({ Data, image }) => {
  return (
    <div className="card" >
      <div className={image} alt="luke"><div className="favorite"></div></div>
      <h3 className="card-header">{Data.name}</h3>
      <h5 className="sub-categories top">Species: <span className="normal">{Data.species}</span></h5>
      <h5 className="sub-categories">Homeworld: <span className="normal">{Data.homeworld}</span></h5>
      <h5 className="sub-categories">Population: <span className="normal">{Data.population}</span></h5>
    </div>
  )
}

export default PeopleCard;