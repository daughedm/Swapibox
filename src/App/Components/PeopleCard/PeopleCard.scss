$sw-yellow: rgb(246, 231, 48);

.card {
  background-color: white;
  width: 250px;
  height: 350px;
  padding: 15px;
  margin: 15px;
  overflow: auto;
}

.card-header {
  font-family: 'Ubuntu', sans-serif;
  font-weight: 700;
  font-size: 1.5rem;
  color: rgb(92, 92, 92);
  border-bottom: 2px solid $sw-yellow;
  margin: 0;
  padding: 15px 10px 15px 10px;
}

.sub-categories {
  font-size: 1rem;
  font-weight: 400;
  color: rgb(92, 92, 92);
  margin: 0;
  padding: 10px 10px 10px 10px;
}

.normal {
  font-weight: 300;
  color: rgb(92, 92, 92);
}

.top {
  padding: 20px 10px 10px 10px;
}

.people0 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/lukeskywalker.jpg);
  background-size: cover;
}

.people1 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/C3PO.jpg);
  background-size: cover;
}

.people2 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/R2D2.jpg);
  background-size: cover;
}

.people3 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/DarthVader.jpg);
  background-size: cover;
}

.people4 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/leia.jpg);
  background-size: cover;
}

.people5 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/owenlars.jpg);
  background-size: cover;
}

.people6 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/beru.jpg);
  background-size: cover;
}

.people7 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/R5D4.jpg);
  background-size: cover;
}

.people8 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/biggs.jpg);
  background-size: cover;
}

.people9 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Obiwan.jpg);
  background-size: cover;
}

.planets0 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/alderaan.jpg);
  background-size: cover;
}

.planets1 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Yavin.jpg);
  background-size: cover;
}

.planets2 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Hoth.jpg);
  background-size: cover;
}

.planets3 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Dagobah.jpg);
  background-size: cover;
}

.planets4 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Bespin.jpg);
  background-size: cover;
}

.planets5 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/endor.jpg);
  background-size: cover;
}

.planets6 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/naboo.jpg);
  background-size: cover;
}

.planets7 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Coruscant.jpg);
  background-size: cover;
}

.planets8 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/kamino.jpg);
  background-size: cover;
}

.planets9 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/geonosis.jpg);
  background-size: cover;
}

.vehicle0 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/sandcrawler.jpg);
  background-size: cover;
}

.vehicle1 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/skyhopper.jpg);
  background-size: cover;
}

.vehicle2 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/landspeeder.jpg);
  background-size: cover;
}

.vehicle3 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/tie-fighter.jpg);
  background-size: cover;
}

.vehicle4 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Snowspeeder.jpg);
  background-size: cover;
}

.vehicle5 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/tie-bomber.jpg);
  background-size: cover;
}

.vehicle6 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/ATAT.jpg);
  background-size: cover;
}

.vehicle7 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/ATST.jpg);
  background-size: cover;
}

.vehicle8 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/twinpod.jpg);
  background-size: cover;
}

.vehicle9 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/sailbarge.jpg);
  background-size: cover;
}