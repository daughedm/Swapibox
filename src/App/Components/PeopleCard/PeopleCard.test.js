import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import PeopleCard from './PeopleCard';

describe('PeopleCard', () => {
  let wrapper;

  it('should match snapshot', () => {
    const person = {
      "name": "Luke Skywalker",
      "homeworld": "Alderaan",
      "species": "Human"
    }
    const addFavorite = jest.fn();
    wrapper = shallow(<PeopleCard item={person}
      Data="person"
      addFavorite={addFavorite} />)

    expect(wrapper).toMatchSnapshot();
  });
})
