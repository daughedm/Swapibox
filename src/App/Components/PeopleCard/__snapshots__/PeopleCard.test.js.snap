// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PeopleCard
    Data="person"
    addFavorite={[Function]}
    item={
        Object {
            "homeworld": "Alderaan",
            "name": "Luke Skywalker",
            "species": "Human",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          alt="person"
          className={undefined}
>
          <div
                    className="favorite"
                    onClick={[Function]}
          >
                    <img
                              alt="favorite"
                              className="fav-icon"
                              src="like.svg"
                    />
          </div>
</div>,
        <h3
          className="card-header"
/>,
        <h5
          className="sub-categories top"
>
          Species:
          <span
                    className="normal"
          />
</h5>,
        <h5
          className="sub-categories"
>
          Homeworld:
          <span
                    className="normal"
          />
</h5>,
        <h5
          className="sub-categories"
>
          Population:
          <span
                    className="normal"
          />
</h5>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "person",
          "children": <div
            className="favorite"
            onClick={[Function]}
>
            <img
                        alt="favorite"
                        className="fav-icon"
                        src="like.svg"
            />
</div>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="favorite"
              className="fav-icon"
              src="like.svg"
/>,
            "className": "favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "favorite",
              "className": "fav-icon",
              "src": "like.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "card-header",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Species:",
            <span
              className="normal"
/>,
          ],
          "className": "sub-categories top",
        },
        "ref": null,
        "rendered": Array [
          "Species:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "normal",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld:",
            <span
              className="normal"
/>,
          ],
          "className": "sub-categories",
        },
        "ref": null,
        "rendered": Array [
          "Homeworld:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "normal",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population:",
            <span
              className="normal"
/>,
          ],
          "className": "sub-categories",
        },
        "ref": null,
        "rendered": Array [
          "Population:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "normal",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "h5",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            alt="person"
            className={undefined}
>
            <div
                        className="favorite"
                        onClick={[Function]}
            >
                        <img
                                    alt="favorite"
                                    className="fav-icon"
                                    src="like.svg"
                        />
            </div>
</div>,
          <h3
            className="card-header"
/>,
          <h5
            className="sub-categories top"
>
            Species:
            <span
                        className="normal"
            />
</h5>,
          <h5
            className="sub-categories"
>
            Homeworld:
            <span
                        className="normal"
            />
</h5>,
          <h5
            className="sub-categories"
>
            Population:
            <span
                        className="normal"
            />
</h5>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "person",
            "children": <div
              className="favorite"
              onClick={[Function]}
>
              <img
                            alt="favorite"
                            className="fav-icon"
                            src="like.svg"
              />
</div>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="favorite"
                className="fav-icon"
                src="like.svg"
/>,
              "className": "favorite",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "favorite",
                "className": "fav-icon",
                "src": "like.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card-header",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Species:",
              <span
                className="normal"
/>,
            ],
            "className": "sub-categories top",
          },
          "ref": null,
          "rendered": Array [
            "Species:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "normal",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld:",
              <span
                className="normal"
/>,
            ],
            "className": "sub-categories",
          },
          "ref": null,
          "rendered": Array [
            "Homeworld:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "normal",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population:",
              <span
                className="normal"
/>,
            ],
            "className": "sub-categories",
          },
          "ref": null,
          "rendered": Array [
            "Population:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "normal",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "h5",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
